samples/v2/subdagio/__init__.py

```diff
@@ -0,0 +1,8 @@
+from subdagio import artifact
```

Comment:
why do we need an `__init__.py` file in the sample folder?

Reply:
This was so that we can isolate/group the tests used for subdag outputs into a module. We can move the tests up level and remove the `subdagio` module if that is the preferred method.

Reply:
So that we can import like [this](https://github.com/kubeflow/pipelines/pull/11196/files/704d349faafd8c21cc978c9dacbda504605de912#diff-aa23d4179ee7858c47fb40841752ade320139212dcb99f9ddb37c9f00127f4f4R30) instead of importing each individual module in `sample_test.py`. If it's preferable, we can remove the init and import the modules individually or take the tests out of the subdirectory like @zazulam suggested.